### DISCUSSION TITLE
Anyone else experiencing cursor flicking in nightly?

### DISCUSSION THREAD

**Discussion by RayGuo-ergou** [ibhagwan/fzf-lua#1596] [General]
After update to recent nightly:
![1](https://github.com/user-attachments/assets/88b700e3-7761-4248-9793-94cd926fde13)

Seems caused by:https://github.com/neovim/neovim/pull/31562 and surprisingly so far I only experience this issue with 'find files', e.g. find commits does not flicking.

**Comment by ibhagwan** [comment]
> Seems caused by:https://github.com/neovim/neovim/pull/31562 and surprisingly so far I only experience this issue with 'find files', e.g. find commits does not flicking.

I’ll check if there’s something that can be done from fzf-lua’a side about this.

**Reply by RayGuo-ergou** [reply]
Thank you!

And I just tried with `mini.sh`, same behavior. 

It has cursor blinking by default (which introduced by the PR I mentioned), a bit hard to notice but you can still see sometime when delete a char, the cursor and the letter overlap.

![GIF 19-12-2024 9-46-03 AM](https://github.com/user-attachments/assets/88da9fe5-9f1c-4a22-a221-d676627e779d)

<hr />

My first guess would be something related this change (mentioned in that PR):
> since terminal windows now use the real cursor

<hr />

Another interesting behavior is that when select an entry, the cursor will reposition to beginning (maybe always do? but at least in the older neovim I cannot notice).

**Reply by ibhagwan** [reply]
> Another interesting behavior is that when select an entry, the cursor will reposition to beginning (maybe always do? but at least in the older neovim I cannot notice).

Do you have line numbers/columns in your grep output. Does the preview show the column?

**Reply by RayGuo-ergou** [reply]
![image](https://github.com/user-attachments/assets/fc307cf3-d534-450e-b61d-4eca054ead89)
Yes I do.

**Reply by ibhagwan** [reply]
> Yes I do.

No, I meant in grep, some users remove `--column` and then don’t have line/columns in their grep output so fzf-lua doesn’t know the line/col, when switching between buffers/files it shouldn’t move the cursor.

**Reply by RayGuo-ergou** [reply]
I am using ripgrep and did not change `rg_opts` in config

**Comment by RayGuo-ergou** [comment]
I am also seeing this in lazygit's commit input, I think it's related to neovim itself or the terminal emulator.

Sorry for the false report.

**Comment by ibhagwan** [comment]
So I looked into it, the merged PR you linked https://github.com/neovim/neovim/pull/31562, it basically proxies your current terminal cursor to neovim, from my short testing the flickering depends on the terminal type, different on alacritty, wezterm, windows term, etc).

The good thing is that the PR also adds another terminal mode setting `t` in `guicursor` so you can control this behavior.

If you wish to restore the previous non blinking block cursor:
```lua
:lua vim.opt.guicursor = "n-v-c-sm:block,i-ci-ve:ver25,r-cr-o:hor20,t:block"
```

If you want a vertical line (similar to INSERT mode):
```lua
:lua vim.opt.guicursor = "n-v-c-sm:block,i-ci-ve:ver25,r-cr-o:hor20,t:ver25"
```

I'm considering making one of the above the default with fzf-lua but I'm not yet sure about it as `guicursor` is a global setting meaning I would have to set and restore it.

**Reply by RayGuo-ergou** [reply]
Thanks for looking into it! I’ve already disabled the blinking cursor ( I set to `n-v-c-sm:block,i-ci-ve:ver25,r-cr-o:hor20` which should do the same thing), but the issue is that the cursor isn't as smooth as it used to be. For example, when inputting or deleting characters, or navigating through options in the file list, I can see it flickering abnormally.

It’s not very noticeable, and in fact, I haven’t noticed it at all in the past two days unless I think about it. I think I can live with it :-).

Thanks for your time and have a good holiday! 🎄

**Reply by ibhagwan** [reply]
Cheers and happy holidays 🎉
